test/lifecycle_module_test.dart

```diff
@@ -2309,6 +2315,148 @@ void runTests(bool runSpanTests) {
     });
   }, timeout: new Timeout(new Duration(seconds: 2)));
 
+  group('without name with children', () {
+    TestLifecycleModule childModule;
+    UnnamedModule parentModule;
+    List<StreamSubscription> subs = [];
+
+    setUp(() async {
+      parentModule = new UnnamedModule();
+      childModule = new TestLifecycleModule(name: 'child');
+
+      subs.add(getTestTracer()
+          .onSpanFinish
+          .where((span) => span.tags['module.name'] != 'child')
+          .listen((span) {
+        fail(
+            'Only the child module should have spans. Found: ${span.tags['module.name']}');
+      }));
+
+      await parentModule.load();
+    });
+
+    tearDown(() async {
+      await parentModule.dispose();
+      await childModule.tearDown();
+      subs.forEach((sub) => sub.cancel());
+      subs.clear();
+    });
+
+    if (runSpanTests) {
+      test('should record a span for child with no parent', () async {
```

Comment:
> for child with no parent

#nit "parent" is ambiguous here. What about something like:
> for child with unnamed parent, and thus no parentContext

Reply:
I can fix this in 1540 which is also in CR right after this one